src/Bicep.Core/Analyzers/Linter/Rules/NoHardcodedEnvironmentUrlsRule.cs

```diff
@@ -170,6 +179,20 @@ public override void VisitStringSyntax(StringSyntax syntax)
                 }
                 base.VisitStringSyntax(syntax);
             }
+
+            public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax syntax)
+            {
+                // Skip resource 'type' strings - there's no reason to perform analysis on them
+                this.VisitNodes(syntax.LeadingNodes);
```

Comment:
>LeadingNodes [](http://example.com/codeflow?start=39&length=12)

Do we need to visit the decorators?

Reply:
`LeadingNodes` contains the decorators - I don't think we should skip all of them. We should consider skipping `@description()`, but I couldn't find an easy way to do it in this PR.

Reply:
You'd probably have to check if the decorator itself is bound to the symbol that represents `@description()`. Makes sense to leave it for now, though.

Reply:
> You'd probably have to check if the decorator itself is bound to the symbol that represents `@description()`. Makes sense to leave it for now, though.

Yeah - but that's not super easy to do today. There's no mechanism to check equality between function symbols AFAIK - we generally rely on string comparisons which isn't super robust.

Reply:
We don't have any other source of truth right for that information. If you check the bound symbol namespace and name against the string values you're expecting, it's sufficient. I'm not really worried about it, though.